app/controllers/faqs_controller.rb

```diff
@@ -0,0 +1,61 @@
+class FaqsController < ApplicationController
+  before_action :require_sign_in, except: :index
+  before_action :authorize_user, except: :index
+
+  def new
+    @faq = Faq.new
+  end
+
+  def create
+    @faq = Faq.new
+    @faq.question = params[:faq][:question]
+    @faq.answer = params[:faq][:answer]
+
+    if @faq.save
+      redirect_to faqs_path
+    else
+      flash[:alert] = 'error saving faq'
+    end
+  end
+
+  def update
+    @faq = Faq.find(params[:id].to_i)
+    @faq.question = params[:faq][:question]
+    @faq.answer = params[:faq][:answer]
+
+    if @faq.save
+      redirect_to faqs_path
+    else
+      flash[:alert] = 'error saving faq'
+    end
+  end
+
+  def edit
+    @faq = Faq.find(params[:id].to_i)
+  end
+
+  def destroy
+    @faq = Faq.find(params[:id].t0_i)
+
+    if @faq.destroy
```

Comment:
these are basically the same as the above except you are doing FAQ instead of events. how could you de-dup this code?